src/actions/breakpoints.js

```diff
@@ -15,30 +15,80 @@ import { getBreakpoint, getBreakpoints, getSource } from "../selectors";
 import type { ThunkArgs } from "./types";
 import type { Location } from "../types";
 
-type addBreakpointOptions = { condition: string, getTextForLine?: () => any };
+type addBreakpointOptions = {
+  condition: string,
+  disabled?: boolean,
+  getTextForLine?: () => any
+};
 
 function _breakpointExists(state, location: Location) {
   const currentBp = getBreakpoint(state, location);
   return currentBp && !currentBp.disabled;
 }
 
-function _getOrCreateBreakpoint(state, location, condition) {
-  return getBreakpoint(state, location) || { location, condition, text: "" };
+function _createBreakpoint(location: Location, opts: Object = {}) {
+  return Object.assign({}, { location }, opts);
+}
+
+async function _getSourceLocation(state, sourceMaps, breakpoint, location) {
```

Comment:
Hmm, what if we changed this to be `_getGeneratedLocation(sourceMaps, breakpoint, source)`

* we change `_getSourceLocation` to `_getGeneratedLocation` to be more clear
* we pass in source instead of state
* we get location from `bp` because passing in location separately is confusing